config/internal/configsource/component.go

```diff
@@ -67,21 +75,33 @@ type Session interface {
 }
 
 // Retrieved holds the result of a call to the Retrieve method of a Session object.
-type Retrieved struct {
+type Retrieved interface {
 	// Value is the retrieved data that will be injected on the configuration.
-	Value interface{}
+	Value() interface{}
+}
+
+// WatchableRetrieved holds the result of a call to the Retrieve method of a Session object
+// for a value that can be watched for updates.
+type WatchableRetrieved interface {
+	Retrieved
+
 	// WatchForUpdate must not return until one of the following happens:
 	//
-	// 1. An update is detected for the monitored value.
+	// 1. An update is detected for the monitored value. In this case the function should
+	//    return ErrValueUpdated or an error wrapping it.
 	//
 	// 2. The parent Session object is closed, in which case the method should return
-	//    ErrSessionClosed.
+	//    ErrSessionClosed or an error wrapping it.
 	//
 	// 3. An error happens while watching for updates. The method should not return
 	//    on first instances of transient errors, optionally there should be
 	//    configurable thresholds to control for how long such errors can be ignored.
 	//
-	// The method must return with a nil error when an update has happened to
-	// the value monitored by the Watcher.
-	WatchForUpdate func() error
+	// This method must only be called when the RetrieveEnd method of the Session that
+	// retrieved the value was successfully completed.
+	WatchForUpdate() error
 }
+
+// WatchForUpdate defines the signature used by the method used to monitor for updates
+// on a retrieved configuration.
+type WatchForUpdate func() error
```

Comment:
No need to have this now, is a simple signature. Limit public API until we need it :)

Reply:
Yeah, it was a convenience for me to experiment not needed now.